lib/opsgenie_alert.rb

```diff
@@ -0,0 +1,93 @@
+require 'httparty'
+require 'active_support/all'
+require 'active_support/time'
+
+class OpsgenieAlert
+
+  attr_reader :alert
+
+  def initialize(alert)
+    @alert = alert
+  end
+
+  def tags
+    alert["tags"]
+  end
+
+  def id
+    alert["id"]
+  end
+
+  def created_at
+    time = time_with_zone(string: alert["createdAt"])
+  end
+
+  def created_at_date
+    alert["createdAt"].to_s.split("T")[0]
+  end
+
+  def saturday?
+    created_at.saturday?
+  end
+
+  def sunday?
+    created_at.sunday?
+  end
+
+  def inhours?
+    time_range_inhours.include?(created_at.hour)
+  end
+
+  def wakinghours_weekday?
+    time_range_wakinghours_weekday.include?(created_at.hour)
+  end
+
+  def wakinghours_weekend?
+    time_range_wakinghours_weekend.include?(created_at.hour)
+  end
```

Comment:
This is more complicated than it needs to be. You know if it's a weekend from the day (create an `on_weekend?` method). You only care here about whether or not it's within the times we consider waking hours (which overlap in hours on weekdays). If you want to exclude the in hours times from the waking hours, using the suggestion above, you could do:

```ruby
def during_out_of_hours_waking_hours?
  return false if in_hours?

  8..23.include?(created_at.hour)
end
```

or 


```ruby
def during_out_of_hours_waking_hours?
  out_of_hours? && 8..23.include?(created_at.hour)
end
```